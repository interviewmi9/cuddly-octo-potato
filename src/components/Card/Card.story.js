import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import Card from './'

const item = {
  id: 12,
  title: 'consequatur autem doloribus',
  description: ' natus consectetur',
  colorId: 2,
}
storiesOf('Card', module)
  .addDecorator((story, context) => withInfo()(story)(context))
  .add('Default view', () => {
    return <Card item={item} remove={() => {}} />
  })
