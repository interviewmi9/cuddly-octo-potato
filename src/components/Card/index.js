import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Spacer = styled.div`
  width: 20%;
`

const Wrapper = styled.div`
  cursor: pointer;
  border-left: 8px solid transparent;
  padding: 20px;
  color: #606b87;
  margin: 20px;
  transition: background 0.4s ease;
  box-shadow: 0px 0px 3px #d4d5d8;
  background-color: #fff;
  &:first-letter {
    text-transform: capitalize;
  }
`

const Card = ({ item }) => {
  return (
    <Spacer>
      <Wrapper title={item.title}>{item.title}</Wrapper>
    </Spacer>
  )
}

Card.propTypes = {
  item: PropTypes.shape(PropTypes.string).isRequired,
}

export default Card
