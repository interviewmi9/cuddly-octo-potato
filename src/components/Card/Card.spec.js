import React from 'react'
import Card from './'

describe('Product Card component', () => {
  const item = {
    id: 12,
    title: 'consequatur autem doloribus natus consectetur',
    colorId: 2,
  }

  it('renders correctly', () => {
    const rendered = render(<Card item={item} />)
    expect(rendered).toMatchSnapshot()
  })
})
