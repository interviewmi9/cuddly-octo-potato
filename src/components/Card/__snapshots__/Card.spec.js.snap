// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card component renders correctly 1`] = `
.c0 {
  width: 20%;
}

.c1 {
  cursor: pointer;
  border-left: 8px solid transparent;
  padding: 20px;
  color: #606b87;
  margin: 20px;
  -webkit-transition: background 0.4s ease;
  transition: background 0.4s ease;
  box-shadow: 0px 0px 3px #d4d5d8;
  background-color: #fff;
}

.c1:first-letter {
  text-transform: capitalize;
}

<div
  className="c0"
>
  <div
    className="c1"
    title="consequatur autem doloribus natus consectetur"
  >
    consequatur autem doloribus natus consectetur
  </div>
</div>
`;
